Podcast/SearchTableViewController.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  SearchTableViewController.swift
+//  Podcast
+//
+//  Created by Kevin Greer on 3/3/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+enum SearchType {
+    case episodes
+    case series
+    case people
+    case tags
+}
+
+protocol SearchTableViewControllerDelegate {
+    func searchTableViewController(controller: SearchTableViewController, didTapSearchResultOfType searchType: SearchType, model: Any)
+}
+
+class SearchTableViewController: UITableViewController {
+    
+    var searchType: SearchType!
+    let cellIdentifiersClasses: [SearchType: (String, AnyClass)] =
+        [.episodes: ("EpisodeCell", SearchEpisodeTableViewCell.self),
+         .series: ("SeriesCell", SearchSeriesTableViewCell.self),
+         .people: ("PeopleCell", SearchPeopleTableViewCell.self),
+         .tags: ("TagCell", SearchTagTableViewCell.self)]
+    let cellHeights: [SearchType: CGFloat] =
+        [.episodes: 84,
+         .series: 95,
+         .people: 76,
+         .tags: 53]
+    
+    var searchResults: [SearchType: [Any]] = [
+        .episodes: [],
+        .series: [],
+        .people: [],
+        .tags: []]
+    
+    var cellDelegate: SearchTableViewControllerDelegate?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let (cellIdentifier, cellClass) = cellIdentifiersClasses[searchType] else { return }
+        tableView.register(cellClass, forCellReuseIdentifier: cellIdentifier)
+        tableView.showsVerticalScrollIndicator = false
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResults[searchType]?.count ?? 0
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return cellHeights[searchType] ?? 0
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let (cellIdentifier, _) = cellIdentifiersClasses[searchType], let results = searchResults[searchType] else { return UITableViewCell() }
+        
+        switch searchType! {
```

Comment:
if you're going to force unwrap searchType, it might as well not be optional to begin with